### PR TITLE
add frontend maxconn and server-state options

### PR DESCRIPTION
**Pull request by MonolithProjects**
This PR is adding:
- option to set `maxconn` on a frontend
- options to store and load server states

### CODE REVIEW

File: templates/etc/haproxy/haproxy-default.cfg.j2 [hunk 1]
```diff
@@ -45,3 +45,6 @@ defaults
 {% if haproxy_default_unique_id_format is defined and haproxy_default_unique_id_format | length %}
     unique-id-format {{ haproxy_default_unique_id_format }}
 {% endif %}
+{% if haproxy_default_load_server_state_from_file is defined and haproxy_default_load_server_state_from_file|length %}
+    load-server-state-from-file {{ haproxy_default_load_server_state_from_file | default('global') }}
```
Comment (goldyfruit) [inline templates/etc/haproxy/haproxy-default.cfg.j2]: Not sure you a need to set `default()` as the statement only apply if `haproxy_default_load_server_state_from_file` is defined.

File: templates/etc/haproxy/haproxy-global.cfg.j2 [hunk 2]
```diff
@@ -76,3 +76,6 @@ global
         {% endfor %}
     {% endfor %}
 {% endif %}
+{% if haproxy_global_server_state_base is defined %}
+    server-state-base {{ haproxy_global_server_state_base | default('current') }}
```
Comment (goldyfruit) [inline templates/etc/haproxy/haproxy-global.cfg.j2]: Not sure you a need to set `default()` as the statement only apply if `haproxy_global_server_state_base` is defined.

**Review by goldyfruit — CHANGES_REQUESTED** [review]

File: templates/etc/haproxy/haproxy-default.cfg.j2 (hunk 1 above)
Comment (MonolithProjects) [reply templates/etc/haproxy/haproxy-default.cfg.j2]: I agree. Will remove it.

File: templates/etc/haproxy/haproxy-global.cfg.j2 (hunk 2 above)
Comment (MonolithProjects) [reply templates/etc/haproxy/haproxy-global.cfg.j2]: I agree. Will remove it.

**Review by goldyfruit — APPROVED** [review]

**Comment by goldyfruit** [comment]
Thanks @MonolithProjects :+1:

**Pull request merged**